x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_initialize_rules_table_saved_state.test.ts

```diff
@@ -0,0 +1,410 @@
+      expect(actions.setPage).not.toHaveBeenCalled();
+      expect(actions.setPerPage).not.toHaveBeenCalled();
+    });
+
+    it('restores only sorting field', () => {
+      mockState({ urlState: null, storageState: { sort: { field: 'name' } } });
+
+      renderHook(() => useInitializeRulesTableSavedState());
+
+      expect(actions.setFilterOptions).toHaveBeenCalledWith(INITIAL_FILTER_OPTIONS);
+      expect(actions.setSortingOptions).toHaveBeenCalledWith({
+        field: 'name',
+        order: INITIAL_SORTING_OPTIONS.order,
+      });
+      expect(actions.setPage).not.toHaveBeenCalled();
+      expect(actions.setPerPage).not.toHaveBeenCalled();
+    });
+
+    it('restores only sorting order', () => {
+      mockState({ urlState: null, storageState: { sort: { order: 'asc' } } });
+
+      renderHook(() => useInitializeRulesTableSavedState());
+
+      expect(actions.setFilterOptions).toHaveBeenCalledWith(INITIAL_FILTER_OPTIONS);
+      expect(actions.setSortingOptions).toHaveBeenCalledWith({
+        field: INITIAL_SORTING_OPTIONS.field,
+        order: 'asc',
+      });
+      expect(actions.setPage).not.toHaveBeenCalled();
+      expect(actions.setPerPage).not.toHaveBeenCalled();
+    });
+
+    it('restores only page number', () => {
+      mockState({ urlState: null, storageState: { page: 10 } });
+
+      renderHook(() => useInitializeRulesTableSavedState());
+
+      expect(actions.setFilterOptions).toHaveBeenCalledWith(INITIAL_FILTER_OPTIONS);
+      expect(actions.setSortingOptions).not.toHaveBeenCalled();
+      expect(actions.setPage).toHaveBeenCalledWith(10);
+      expect(actions.setPerPage).not.toHaveBeenCalled();
+    });
+
+    it('restores only page size', () => {
+      mockState({ urlState: null, storageState: { perPage: 10 } });
+
+      renderHook(() => useInitializeRulesTableSavedState());
+
+      expect(actions.setFilterOptions).toHaveBeenCalledWith(INITIAL_FILTER_OPTIONS);
+      expect(actions.setSortingOptions).not.toHaveBeenCalled();
+      expect(actions.setPage).not.toHaveBeenCalled();
+      expect(actions.setPerPage).toHaveBeenCalledWith(10);
+    });
+  });
+
+  describe('when state is saved in the url and the storage', () => {
+    beforeEach(() => {
+      mockState({ urlState: urlSavedState, storageState: storageSavedState });
+    });
+
+    it('restores the state from the url', () => {
+      renderHook(() => useInitializeRulesTableSavedState());
+
+      expect(actions.setFilterOptions).toHaveBeenCalledWith({
+        filter: urlSavedState.searchTerm,
+        showCustomRules: urlSavedState.showCustomRules,
+        showElasticRules: !urlSavedState.showCustomRules,
+        tags: urlSavedState.tags,
+      });
+      expect(actions.setSortingOptions).toHaveBeenCalledWith(urlSavedState.sort);
+      expect(actions.setPage).toHaveBeenCalledWith(urlSavedState.page);
+      expect(actions.setPerPage).toHaveBeenCalledWith(urlSavedState.perPage);
+    });
+  });
```

Comment:
Maybe an additional test case to test that in case of partial urlState and storageState, some pieces can be restored from URL and others from storage?

Reply:
Good point, I've added such a test.